far/editor.cpp

```diff
@@ -3395,6 +3395,100 @@ int Editor::CalculateSearchNextPositionInTheLine(const bool Backward, const bool
 	}
 }
 
+namespace
+{
+	[[nodiscard]] short radix10_formatted_width(const size_t num)
+	{
+		return static_cast<short>(std::log10(num)) + 1;
+	}
+
+	class find_all_list
+	{
+	public:
+		find_all_list(int MaxLinesCount)
+			: m_LineNumColumnMaxWidth{ radix10_formatted_width(MaxLinesCount) }
+		{}
+
+		void add_item(FindCoord FoundCoords, const string& ItemText)
+		{
+			MenuItemEx Item{ far::format(L"{:{}}{:{}}{}"sv,
+				FoundCoords.Line + 1, m_LineNumColumnMaxWidth,
+				FoundCoords.Pos + 1, m_FoundPosColumnMaxWidth,
+				ItemText) };
+			Item.Annotations.emplace_back(FoundCoords.Pos, segment::length_tag{ FoundCoords.SearchLen });
+			Item.ComplexUserData = FoundCoords;
+			m_Menu->AddItem(Item);
+
+			if (FoundCoords.Line != m_LastSeenLine)
+			{
+				m_LastSeenLine = FoundCoords.Line;
+				m_UniqueLineCount++;
+			}
+			m_MaxLineNum = std::max(m_MaxLineNum, FoundCoords.Line);
+			m_MaxFoundPos = std::max(m_MaxFoundPos, FoundCoords.Pos);
+		}
+
+		void make_ready()
+		{
+			m_MenuY2 = std::min(ScrY, m_MenuY1 + std::min(static_cast<int>(m_Menu->size()), 10) + 2);
+
+			m_Menu->SetMenuFlags(VMENU_WRAPMODE | VMENU_SHOWAMPERSAND | VMENU_ENABLEALIGNANNOTATIONS);
+			m_Menu->SetPosition({ -1, m_MenuY1, 0, m_MenuY2 });
+			m_Menu->SetTitle(far::vformat(msg(lng::MEditSearchStatistics), m_Menu->size(), m_UniqueLineCount));
+			m_Menu->SetBottomTitle(KeysToLocalizedText(KEY_CTRLENTER, KEY_F5, KEY_ADD, KEY_CTRLUP, KEY_CTRLDOWN));
+			m_Menu->SetHelp(L"FindAllMenu"sv);
+			m_Menu->SetId(EditorFindAllListId);
+
+			const short LineNumColumnWidth{ radix10_formatted_width(m_MaxLineNum) };
+			const short FoundPosColumnWidth{ radix10_formatted_width(m_MaxFoundPos) };
+			const short LineNumColumnStart{ m_LineNumColumnMaxWidth - LineNumColumnWidth };
+			const short FoundPosColumnStart{ m_LineNumColumnMaxWidth + m_FoundPosColumnMaxWidth - FoundPosColumnWidth };
+			const short ItemTextStart{ m_LineNumColumnMaxWidth + m_FoundPosColumnMaxWidth };
+			m_Menu->SetFixedColumns(
+				{
+					{
+						.TextSegment{ LineNumColumnStart, small_segment::length_tag{ LineNumColumnWidth } },
+						.CurrentWidth{ LineNumColumnWidth },
```

Comment:
Clang seems to be unhappy with curly braces around scalar initializer. Is it possible to enable this syntax [-Wbraced-scalar-init]. If not, I'll fix the code.

Reply:
Actually, replaced curly braces with `=`.